gigascience/gigadb-website
# Remove need for temporary and insecure hack to out dated dependency

# Pull request for issue: #1626

THis PR should be merged after https://github.com/gigascience/gigadb-website/pull/1678

This is a pull request for the following functionalities:

* Update node v to get rid of legacy openssl
* non-breaking update of vue widget deps

## How to test?

* Run project locally regenerating `.env` file from `env-sample` (or just change node version in your `.env` file to `NODE_VERSION=20.11.0` and verify that it works without using the `--openssl-legacy-provider` option
* Run gitlab CI/CD pipelines and cross check that they work as expected (cannot run these pipelines myself)

## How have functionalities been implemented?

The `--openssl-legacy-provider` option was necessary with node 17, so after upgrading node is no longer useful

- Of all the dependencies of teh Vue widget which present vulnerabilities, upgraded those which did not cause build errors: axios, karma and json-server
- Set `NODE_VERSION=20.11.0` and removed `--openssl-legacy-provider` from docker-compose file

NOTE: I updated node version across the board, can you verify that node is used only for the FUW widget? Judging by the docker files present, this looks like the case, but not 100% sure. @rija ?

NOTE: Some dependency updates caused build errors or cannot be updated (`npm audit` shows which ones)
* Webpack v4 to v5: causes breaking errors and would require delving a bit deeper into the config
* vue-loader: cannot be upgraded with Vue 2 (latest version needs Vue 3)
* css-loader: latest version needs webpack v5

## Any issues with implementation?

* Unable to test the gitlab pipelines

## Any changes to automated tests?

--

## Any changes to documentation?

--

## Any technical debt repayment?

--

## Any improvements to CI/CD pipeline?

* Updated node version
